pkg/provider/github/github.go

```diff
@@ -450,3 +450,34 @@ func ListRepos(ctx context.Context, v *Provider) ([]string, error) {
 	}
 	return repoURLs, nil
 }
+
+func (v *Provider) ListRepository(ctx context.Context, repository []string, run *params.Run, event *info.Event) (string, error) {
```

Comment:
remind me what is this function is supposed to do ? would that work with github token ? can it be in SetClient instead of a new toplevel function on the interface?

Reply:
Actually i am scoping token to list of provided Repos but this is very specific to GithubApp

Planning to takeout this function from Provider and add part of the logic in match.go file itself

Reply:
Actually i am scoping token to list of provided Repos and this is very specific to GithubApp

Planning to name function as `ScopeGithubTokenToListOfRepos`

`SetClient` does specific work if i add scoping token also in `SetClient` it may mess up :thinking: